### PR TITLE
fix(iceberg): fix decimal type for iceberg source

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- It seems this is a bug introduced by a refactoring before

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/src/array/arrow/arrow_impl.rs [hunk 1]
```diff
@@ -522,6 +522,7 @@ pub trait FromArrow {
             Int16 => self.from_int16_array(array.as_any().downcast_ref().unwrap()),
             Int32 => self.from_int32_array(array.as_any().downcast_ref().unwrap()),
             Int64 => self.from_int64_array(array.as_any().downcast_ref().unwrap()),
+            Decimal128(_, _) => self.from_decimal128_array(array.as_any().downcast_ref().unwrap()),
             Decimal256(_, _) => self.from_int256_array(array.as_any().downcast_ref().unwrap()),
```
Comment (xxchan) [inline src/common/src/array/arrow/arrow_impl.rs:526]: why decimal 128 and 256 are different? It is from_int256 below 🤔

Comment (xxchan) [reply src/common/src/array/arrow/arrow_impl.rs:526]: seems a typo for 256

**Review by xxchan — APPROVED** [review]
rubber stamp

File: src/common/src/array/arrow/arrow_impl.rs (hunk 1 above)
Comment (wangrunji0408) [reply src/common/src/array/arrow/arrow_impl.rs:526]: Arrow decimal256 type is used for RisingWave int256 type. 🤔

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**